### PR TITLE
Workout Plan Page

### PR DESCRIPTION
**Pull request by simekanidvt**
*  Workout Plan Page which will lead from the landing page done with mock data
* Collection view redone on the landing page and reverted to a conventional collection view
* Changed my folder structure to separate it by features

Screenshot 2022-03-31 at 11 39 03

### CODE REVIEW

File: WorkoutManager/WorkoutManager/LandingPage/WorkoutManagerViewController.swift [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureCollectionView()
+        viewModel.getWorkoutPlans()
+    }
+}
+
+// MARK: Workout Manager Delegate
+extension WorkoutManagerViewController {
```
Comment (bgouws) [inline WorkoutManager/WorkoutManager/LandingPage/WorkoutManagerViewController.swift:26]: Would be a little bit neater to declare this like this:

`extension WorkoutManagerViewController: WorkoutManagerDelegate {`

File: WorkoutManager/WorkoutManager/LandingPage/WorkoutManagerViewController.swift [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+import UIKit
+
+class WorkoutManagerViewController:UIViewController, WorkoutManagerDelegate {
+    
+    @IBOutlet private var workoutPlansCollectionView : UICollectionView!
+    
+    static let identifier = "WorkoutManagerViewController"
+    
+    private lazy var viewModel = WorkoutManagerViewModel(delegate: self,repository: WorkoutManagerRepository())
```
Comment (bgouws) [inline WorkoutManager/WorkoutManager/LandingPage/WorkoutManagerViewController.swift:16]: Small code style think but put these parameters on their own line 😄

File: WorkoutManager/WorkoutManager/WorkoutPlanPage/CustomCells/WorkoutPlanItem.swift [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    static var identifier = "WorkoutPlanItem"
+    
+    static func nib() -> UINib {
+        return UINib(nibName: "WorkoutPlanItem", bundle: nil)
+    }
+}
+
+extension WorkoutPlanItem: UITableViewDelegate, UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 8
```
Comment (bgouws) [inline WorkoutManager/WorkoutManager/WorkoutPlanPage/CustomCells/WorkoutPlanItem.swift:28]: So normally hardcoded values are a warning sign, also is this a UITableView within a cell?

File: WorkoutManager/WorkoutManager/WorkoutPlanPage/WorkoutPlansViewController.swift [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        super.viewDidLoad()
+        configureCollectionView()
+    }
+    
+    private var currentPage = 0
+}
+
+extension WorkoutPlansViewController:UICollectionViewDelegate, UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return 5
```
Comment (bgouws) [inline WorkoutManager/WorkoutManager/WorkoutPlanPage/WorkoutPlansViewController.swift:34]: This should also be a warning sign, unless you are still making use of some mock data?

File: WorkoutManager/WorkoutManager/WorkoutPlanPage/CustomCells/WorkoutPlanItem.swift (hunk 3 above)
Comment (bgouws) [reply WorkoutManager/WorkoutManager/WorkoutPlanPage/CustomCells/WorkoutPlanItem.swift:28]: So I had a look at your screenshots 😄 thanks for including them 👍 if you are using mock data for now that is fine, but obviously once we have actual data we can expect this to be much more dynamic.

**Review by bgouws — CHANGES_REQUESTED** [review]
Just some cleaning looks pretty solid though 👏

File: WorkoutManager/WorkoutManager/WorkoutPlanPage/CustomCells/WorkoutPlanItem.swift (hunk 3 above)
Comment (simekanidvt) [reply WorkoutManager/WorkoutManager/WorkoutPlanPage/CustomCells/WorkoutPlanItem.swift:28]: Perfect, When i include my API this will change

**Review by bgouws — APPROVED** [review]

**Review by kirsty14 — APPROVED** [review]
Looking good 👍🏻

**Pull request merged**